server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,581 @@
+                                    Hierarchy.MISSING_PREFIX_FILES, controller, path
+                                )
+                            else:
+                                f = 0
+                                try:
+                                    with prefix_file.open(mode="r") as file:
+                                        prefix = file.read().replace("\n", "")
+                                except Exception:
+                                    f = 1
+                                if f == 1:
+                                    rhier.add_unexpected_entries(
+                                        Hierarchy.BAD_PREFIX_FILES, controller, path
+                                    )
+                                else:
+                                    if prefix != prefix_path:
+                                        rhier.add_unexpected_entries(
+                                            Hierarchy.BAD_PREFIXES, controller, path
+                                        )
+                        verify_user_arg(rhier, mdlogcfg, controller, path)
+    return 0
+
+
+# Controller Hierarchy
+def verify_controllers(hier):
+    """ 'hier' is a ControllerHierarchy class object. """
+    controllers = Path(hier.path).glob("*")
+    verify_valid_controllers(hier, controllers)
+
+    for controller in hier.controllers:
+        dir_path = Path(hier.path, controller)
+        unexpected_dirs = list()
+        if not Path(hier.config.ARCHIVE, controller).is_dir():
+            """We have a controller in the hierarchy which does not have a
+            controller of the same name in the archive hierarchy.  All
+            we do is report it, don't bother analyzing it further.
+            """
+            hier.add_controller_list(Hierarchy.MIALIST, Path(controller).name)
+        else:
+            """Report any controllers with objects other than directories
+            and links, while also recording any empty controllers.
+            """
+            count_dir_entries = 0
+            direct_entries = dir_path.glob("*")
+            for item_path in direct_entries:
+                count_dir_entries += 1
+                if not item_path.is_dir() and not item_path.is_symlink():
+                    unexpected_dirs.append(controller)
+                    break
+            else:
+                if count_dir_entries == 0:
+                    hier.add_controller_list(Hierarchy.EMPTY_CONTROLLERS, controller)
+                    continue
+
+            if unexpected_dirs:
+                hier.add_controller_list(Hierarchy.UNEXPECTED_CONTROLLERS, controller)
+            hier.add_verify_list(controller)
+
+    return 0
+
+
+# User Hierarchy
+def verify_users(hier):
+    """ 'hier' is UserHierarchy class object."""
+    if not Path(hier.path).is_dir():
+        print(
+            f"The setting for USERS in the config file is {hier.path}, but that is"
+            f" not a directory"
```

Comment:
You don't need the `f` on this part of the string, since there are no format directives in it.

Reply:
right